x-pack/test/functional/services/ml/data_visualizer_table.ts

```diff
@@ -133,17 +153,119 @@ export function MachineLearningDataVisualizerTableProvider(
+        if (await testSubjects.exists(action)) {
+          await testSubjects.click(this.rowSelector(fieldName, 'euiCollapsedItemActionsButton'));
+          await testSubjects.missingOrFail(action, { timeout: 5000 });
+        }
+      });
+    }
+
+    public async assertActionMenuViewInLensEnabled(fieldName: string, expectedValue: boolean) {
+      await retry.tryForTime(30 * 1000, async () => {
+        await this.ensureActionsMenuOpen(fieldName);
+        const actionMenuViewInLensButton = await find.byCssSelector(
+          '[data-test-subj="dataVisualizerActionViewInLensButton"][class="euiContextMenuItem"]'
+        );
+        const isEnabled = await actionMenuViewInLensButton.isEnabled();
+        expect(isEnabled).to.eql(
+          expectedValue,
+          `Expected "Explore in lens" action menu button for '${fieldName}' to be '${
+            expectedValue ? 'enabled' : 'disabled'
+          }' (got '${isEnabled ? 'enabled' : 'disabled'}')`
+        );
+      });
+    }
+
+    public async assertActionMenuDeleteIndexPatternFieldButtonEnabled(
+      fieldName: string,
+      expectedValue: boolean
+    ) {
+      await this.ensureActionsMenuOpen(fieldName);
+      const actionMenuViewInLensButton = await find.byCssSelector(
+        '[data-test-subj="dataVisualizerActionDeleteIndexPatternFieldButton"][class="euiContextMenuItem"]'
+      );
+      const isEnabled = await actionMenuViewInLensButton.isEnabled();
+      expect(isEnabled).to.eql(
+        expectedValue,
+        `Expected "Delete index pattern field" action menu button for '${fieldName}' to be '${
+          expectedValue ? 'enabled' : 'disabled'
+        }' (got '${isEnabled ? 'enabled' : 'disabled'}')`
+      );
+    }
+    public async clickActionMenuDeleteIndexPatternFieldButton(fieldName: string) {
+      const testSubj = 'dataVisualizerActionDeleteIndexPatternFieldButton';
+      await retry.tryForTime(5000, async () => {
+        await this.ensureActionsMenuOpen(fieldName);
+
+        const button = await find.byCssSelector(
+          '[data-test-subj="dataVisualizerActionDeleteIndexPatternFieldButton"][class="euiContextMenuItem"]'
+        );
+        await button.click();
+        await this.ensureActionsMenuClosed(fieldName, testSubj);
+      });
+    }
+
+    public async assertActionMenuExists(fieldName: string) {
+      const actionButton = this.rowSelector(fieldName, 'euiCollapsedItemActionsButton');
+      await testSubjects.existOrFail(actionButton);
+    }
+
+    public async assertViewInLensActionEnabled(fieldName: string, expectedValue: boolean) {
       const actionButton = this.rowSelector(fieldName, 'dataVisualizerActionViewInLensButton');
       await testSubjects.existOrFail(actionButton);
-      await testSubjects.isEnabled(actionButton);
+      const isEnabled = await testSubjects.isEnabled(actionButton);
+      expect(isEnabled).to.eql(
+        expectedValue,
+        `Expected "Edit index pattern" button for '${fieldName}' to be '${
```

Comment:
The message doesn't match the assertion in this method:
```suggestion
        `Expected "Explore in lens" button for '${fieldName}' to be '${
```

Reply:
Updated here https://github.com/elastic/kibana/pull/101316/commits/6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e